lib/DidJwtStrategy.js

```diff
@@ -0,0 +1,204 @@
+    }
+    req.user = {
+      identity: {id: did}
+    };
+    self.success(req.user);
+  });
+};
+
+Strategy.prototype.setStore = function(store, callback) {
+  this.store = store;
+  async.parallel([
+    callback => {
+      // the store does not handle verification of RSA signed tokens
+      const opts = bedrock.util.clone(config['authn-did-jwt'].namespaces.rsa);
+      delete opts.verify;
+      store.provision('RSA', opts, callback);
+    },
+    callback => store.provision(
+      'HMAC', config['authn-did-jwt'].namespaces.hmac, callback)
+  ], callback);
+};
+
+/**
+ * Verify a JWT signed with a RSA algorithm.
+ *
+ * @param token the token to be verified.
+ *
+ * @param callback(err, tokenPayload) called once the operation completes.
+ */
+Strategy.prototype.verifyRsa = function(token, callback) {
+  let decodedToken;
+  let header;
+
+  try {
+    decodedToken = jwt.decode(token, {complete: true});
+    header = decodedToken.header;
+  } catch(e) {
+    return callback(e);
+  }
+
+  if(!header.kid || !(/^(did:|https:)/.test(header.kid))) {
+    return callback(new Error('Invalid key identifier in token.'));
+  }
+  return async.auto({
+    // Dereference the `kid` URL and find the public key
+    getKey: callback => {
+      if((/^https:/.test(header.kid))) {
+        return request.get(header.kid, (err, res, body) =>
+          callback(err, body));
+      }
+      // kid is a DID URL
+      didio.get(header.kid, callback);
+    },
+    // If publicKey.owner is not `sub` of the JWT, get the DDO for `sub`
+    // and make sure publicKey.owner matches DDO’s `idp` entry and that
+    // it is in the whitelist
+    getDdo: ['getKey', (callback, results) => {
+      // `sub` is guaranteed to be a DID
+      if(results.getKey.owner === decodedToken.payload.sub) {
+        // FIXME: what happens in this case?  Can the sub own the key signing
+        // the token?
+        return callback(new Error('What to do!?'));
+      }
+      if(results.getKey.owner !== decodedToken.payload.sub) {
+        // get DDO for `sub`
+
+        didio.getDidDocument(decodedToken.payload.sub, (err, result) => {
+          if(err) {
+            return callback(err);
+          }
+          // ensure that the key owner is the subject's IdP and the IdP is
+          // whitelisted
+          if(result.idp !== results.getKey.owner ||
+            !_isTrustedSigner(result.idp)) {
+            return callback(new Error(
+              'Token is not signed by a trusted source.'));
+          }
+          // Dereference the URL in publicKey.owner and check that document
+          // for a publicKey that matches `kid` URL
+          didio.getDidDocument(result.idp, (err, result) => {
+            if(err) {
+              return callback(err);
+            }
+            if(!result.publicKey.some(key => key.id === header.kid)) {
+              return callback(new Error('Invalid key specified in token.'));
+            }
+            callback();
+          });
```

Comment:
We should probably move some of the above out into some functions that are named after what's going on... like "checkBidirectionalLink" or "ensureSignedByIdP" ... things like that. It may be easier to understand how verification and trust are established that way.